src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/activities/ActivitiesApiService.kt

```diff
@@ -16,13 +18,23 @@ class ActivitiesApiService(@Qualifier("activitiesApiWebClient") private val webC
     .uri("/migrate/activity")
     .bodyValue(migrateRequest)
     .retrieve()
-    .bodyToMono(ActivityMigrateResponse::class.java)
-    .awaitSingle()!!
+    .bodyToMono<ActivityMigrateResponse>()
+    .awaitSingle()
 
   suspend fun migrateAllocation(migrateRequest: AllocationMigrateRequest): AllocationMigrateResponse = webClient.post()
     .uri("/migrate/allocation")
     .bodyValue(migrateRequest)
     .retrieve()
-    .bodyToMono(AllocationMigrateResponse::class.java)
-    .awaitSingle()!!
+    .bodyToMono<AllocationMigrateResponse>()
+    .awaitSingle()
+
+  suspend fun moveActivityStartDates(prisonCode: String, newStartDate: LocalDate): List<String> = webClient.post()
+    .uri {
+      it.path("/migrate/{prisonCode}/move-activity-start-dates")
+        .queryParam("activityStartDate", newStartDate)
+        .build(prisonCode)
+    }
+    .retrieve()
+    .bodyToMono<List<String>>()
+    .awaitSingle()
```

Comment:
The main change is a new endpoint that moves the start date of all activities for a prison. This is intended to be called after a migration has been run but the prison needs more time before switching from NOMIS to DPS activities.